File: my-folder/1039-find-the-town-judge/solution.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        graph = {i:[] for i in range(1,n+1)}

        for u,v in trust:
            graph[u].append(v)
        
        ans  = -1
        possibles = []
        for u,v in graph.items():
            print(f'u: {u}, v:{v}')
            if len(v)==0:
                possibles.append(u)
        if len(possibles)>1 or len(possibles)==0:
            return -1
        else:
            ans = possibles[0]
            for i in range(1,n+1):
                if i==ans:
                    continue
                else:
                    if ans not in graph[i]:
                        return -1
            return ans

File: my-folder/1039-find-the-town-judge/solution.cpp
class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust) {
        

        vector<int> in_degree(n,0);
        vector<int> out_degree(n,0);

        for (auto &edge:trust)
        {
            int first = edge[0];
            int second = edge[1];
            in_degree[second-1]+=1;
            out_degree[first-1]+=1;
        }

        for (int i=0; i<n; i++)
        {
            if (in_degree[i] ==n-1 and out_degree[i]==0)
            {
                return i+1;
            }

        }
        return -1;


    }
};
